src/o365/spo/commands/hubsite/hubsite-list.ts

```diff
@@ -47,18 +68,103 @@ class SpoHubSiteListCommand extends SpoCommand {
 
         return request.get(requestOptions);
       })
-      .then((res: { value: HubSite[] }): void => {
+      .then((res: { value: HubSite[] }): Promise<any[]|void>=> {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        hubSites = res.value;
+
+        if (args.options.includeAssociatedSites !== true || args.options.output !== 'json') {
+          return Promise.resolve();
+        } else {
+          if (this.debug) {
+            cmd.log('Retrieving associated sites...');
+            cmd.log('');
+          }
+        }
+
+        const requestOptions: any = {
+          url: `${auth.site.url}/_api/web/lists/GetByTitle('DO_NOT_DELETE_SPLIST_TENANTADMIN_AGGREGATED_SITECOLLECTIONS')/RenderListDataAsStream`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata=nometadata'
+          }),
+          json: true,
+          body: {
+            parameters: {
+              ViewXml: "<View><Query><Where><And><And><IsNull><FieldRef Name=\"TimeDeleted\"/></IsNull><Neq><FieldRef Name=\"State\"/><Value Type='Integer'>0</Value></Neq></And><Neq><FieldRef Name=\"HubSiteId\"/><Value Type='Text'>{00000000-0000-0000-0000-000000000000}</Value></Neq></And></Where><OrderBy><FieldRef Name='Title' Ascending='true' /></OrderBy></Query><ViewFields><FieldRef Name=\"Title\"/><FieldRef Name=\"SiteUrl\"/><FieldRef Name=\"SiteId\"/><FieldRef Name=\"HubSiteId\"/></ViewFields><RowLimit Paged=\"TRUE\">" + this.batchSize + "</RowLimit></View>",
+              DatesInUtc: true
+            }
+          }
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        if (this.debug || this.verbose) {
+          cmd.log(`Will retrieve associated sites (including the hub sites) in batches of ${this.batchSize}`);
+        }
+        const getSites = async (reqOptions: any, nonPagedUrl: string, sites: any[] = [], batchNumber: number = 0): Promise<any> => {
```

Comment:
We should avoid using `any[]` in favor of known types. This will help us ensure code integrity in the long run